frontend/src/metabase/hoc/Routeless.jsx

```diff
@@ -41,13 +41,22 @@ export default (ComposedComponent) => class extends Component {
         // if the state previously was the saved one and is now not, then we probably
         // hit the back button, so close the wrapped component
         if (location.state === this._state && nextLocation.state !== this._state) {
-            this.props.onClose();
+            // close in a timeout in case it will be closed anyway and the URL changes.
```

Comment:
Didn't really understand based on this comment that _why_ this solves the described issue (or what the version without timeout was lacking)